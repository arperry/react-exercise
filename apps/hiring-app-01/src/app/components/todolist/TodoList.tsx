import { useState, useEffect } from "react";
import { Container } from "@mantine/core";
import TodoCard from "./TodoCard";
import { getAllTodo } from "app/api";
import { Todo } from "app/types/todo";
import { toast } from "react-toastify";

const TodoList = () => {
  const [todoList, setTodoList] = useState<Todo[]>([]);
  const [todoListUpdated, setTodoListUpdated] = useState<boolean>(false);

  useEffect(() => {
    const fetchTodos = async () => {
      const response = await getAllTodo();
      if (Array.isArray(response.data)) {
        setTodoList(response.data);
        setTodoListUpdated(false);
      } else {
        toast.error("Error loading todo list");
      }
    };
    fetchTodos();
  }, [todoListUpdated]);

  return (
    <Container>
      {todoList.length
        ? todoList.map((todo) => (
            <TodoCard
              setTodoListUpdated={setTodoListUpdated}
              completed={todo.completed}
              title={todo.title}
              duration={todo.duration}
              uuid={todo.uuid}
              key={todo.uuid}
            />
          ))
        : "Nothing Todo"}
    </Container>
  );
};

export default TodoList;
