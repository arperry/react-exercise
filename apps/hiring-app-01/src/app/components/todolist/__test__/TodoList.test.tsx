import renderer from 'react-test-renderer';
import TodoList from '../TodoList';
import { render } from '@testing-library/react';

describe('TodoList', () => {
  const todoList = renderer.create(<TodoList />);

  it('renders correctly', () => {
    expect(todoList.toJSON()).toMatchSnapshot();
  });

  it("contains 'Nothing Todo' if there are no cards", () => {
    const {queryByText} = render(<TodoList />);
    expect(queryByText('Nothing Todo'))
  })
})