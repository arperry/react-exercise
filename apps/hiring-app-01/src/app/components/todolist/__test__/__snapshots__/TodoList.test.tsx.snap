// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList renders correctly 1`] = `
<div
  className="mantine-Container-root mantine-15pjuqq"
>
  Nothing Todo
</div>
`;
